lib/idn.h

```diff
@@ -24,9 +24,16 @@
  *
  ***************************************************************************/
 
+/* for macOS and iOS targets */
+#if defined(__APPLE__)
+#if __has_include(<unicode/uidna.h>)
```

Comment:
Thinking of the gcc (or possibly other non-Apple) compilers targeting Apple, enclosing this in `#if defined __has_include` might be useful to detect availability of the preprocessor feature:
https://gcc.gnu.org/onlinedocs/cpp/_005f_005fhas_005finclude.html

(at the moment GCC 13 is incompatible with SDK 13.x. Haven't tried 15.x though.)

Reply:
Thanks for the idea. I've added a check for whether __has_include is available.

Reply:
LGTM, thanks!